### PR TITLE
feat: resolve function and tests

### LINKED ISSUE #21: implement handle registry `resolve` function
The `resolve` function returns the profile address linked to a handle NFT.

Here are some resources to help you with implementation:
- [Lens implementation](https://github.com/lens-protocol/core/blob/master/contracts/namespaces/TokenHandleRegistry.sol#L212)
- [Handle Documentation](https://docs.lens.xyz/docs/handle)

you are expected to:
- Implement the `resolve` function
- write the associated test for the implemented function

Note that:
- your implementation, doesn't need to be an exact copy of lens implementation. Be flexible, and properly comment your codes.
- Ask questions in the working group, if lost or unsure of anything.

### PR DESCRIPTION
**Pull request by MarvyNwaokobia**
Implemented the handle_registry resolve function and wrote the equivalent tests

resolves #21

### CODE REVIEW

File: src/base/types.cairo [hunk 1]
```diff
@@ -139,3 +139,18 @@ pub struct QuoteParams {
     pointed_profile_address: ContractAddress,
     pointed_pub_id: u256
 }
+
+mod RegistryTypes {
+    use starknet::ContractAddress;
+    #[derive(Drop, Serde, starknet::Store, Clone)]
+    pub struct Token {
```
Comment (Darlington02) [inline src/base/types.cairo:146]: Let's not complicate it. You can take out the types

File: src/base/types.cairo [hunk 2]
```diff
@@ -139,3 +139,18 @@ pub struct QuoteParams {
...
+        id: u256,
+        collection: ContractAddress
+    }
+
+    #[derive(Drop, Serde, starknet::Store, Clone)]
+    pub struct Handle {
+        id: u256,
```
Comment (Darlington02) [inline src/base/types.cairo:153]: take out this too.

File: src/interfaces/IHandleRegistry.cairo [hunk 3]
```diff
@@ -12,6 +12,6 @@ pub trait IHandleRegistry<TState> {
     // *************************************************************************
     //                              GETTERS
     // *************************************************************************
-    fn resolve(self: @TState, handle_id: u256) -> ContractAddress;
+    fn resolve(ref self: TState, handle_id: u256) -> ContractAddress;
```
Comment (Darlington02) [inline src/interfaces/IHandleRegistry.cairo:15]: the `resolve` function is not meant to be an external function ie it shouldn't mutate the state of the contract. Please use the initial interface signature which passed in the state as a snapshot instead.

File: src/namespaces/handle_registry.cairo [hunk 4]
```diff
@@ -79,9 +81,15 @@ mod HandleRegistry {
         // *************************************************************************
         //                            GETTERS
         // *************************************************************************
-        fn resolve(self: @ContractState, handle_id: u256) -> ContractAddress {
-            // TODO
-            0.try_into().unwrap()
+        fn resolve(ref self: ContractState, handle_id: u256) -> ContractAddress {
+            let isExist = IHandleDispatcher { contract_address: self.handle_address.read() }
```
Comment (Darlington02) [inline src/namespaces/handle_registry.cairo:85]: I'd prefer to use `it_exists` for the variable name instead.

File: src/namespaces/handle_registry.cairo [hunk 5]
```diff
@@ -79,9 +81,15 @@ mod HandleRegistry {
...
+                .exists(handle_id);
+            assert(isExist, 'Handle ID does not exist');
+            let resolved_handle_profile_address: ContractAddress = self
+                ._resolve_handle_to_profile_address(
```
Comment (Darlington02) [inline src/namespaces/handle_registry.cairo:89]: No need to over-complicate things by delegating to an internal function. just simply read from the storage state and return the value.

File: src/namespaces/handle_registry.cairo [hunk 6]
```diff
@@ -79,9 +81,15 @@ mod HandleRegistry {
...
+                    RegistryTypes::Handle { id: handle_id, collection: self.handle_address.read() }
```
Comment (Darlington02) [inline src/namespaces/handle_registry.cairo:90]: we don't need a RegistryType. just read `handle_to_profile_address` and return the value.

File: src/namespaces/handle_registry.cairo [hunk 7]
```diff
@@ -139,5 +147,11 @@ mod HandleRegistry {
                     }
                 )
         }
+
+        fn _resolve_handle_to_profile_address(
+            ref self: ContractState, handle: RegistryTypes::Handle
+        ) -> ContractAddress {
+            self.handle_to_profile_address.read(handle.id)
+        }
```
Comment (Darlington02) [inline src/namespaces/handle_registry.cairo:155]: Finally, format your contracts by running `scarb fmt` before pushing your codes else the workflows will fail

**Review by Darlington02 — CHANGES_REQUESTED** [review]

File: src/namespaces/handle_registry.cairo (hunk 7 above)
Comment (MarvyNwaokobia) [reply src/namespaces/handle_registry.cairo]: Done

**Pull request merged**